src/components/authentication/configureFirebase.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState, useCallback } from 'react';
+
+const AuthContext = React.createContext(
+    {
+        token: '',
+        isLoggedIn: true,
+        login: (token) => { },
+        logout: () => { },
+    }
+);
+
+export const AuthContextProvider = (props) => {
+    let initialToken;
+
+    const [token, setToken] = useState(initialToken);
+
+    const userIsLoggedIn = !!token;
+
+    const logoutHandler = useCallback(() => {
+        setToken(null);
+        localStorage.removeItem('token');
+    }, []);
+
+    const loginHandler = (token) => {
+        setToken(token);
+        localStorage.setItem('token', token);
```

Comment:
what do you mean by `localStorage`?

Reply:
Is a web storage object to store the data on the user’s computer locally, which means the stored data is saved across browser sessions and the data stored has no expiration time.

// store user details and jwt token in local storage to keep the user logged in between page refreshes.